Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientUnsecuredTest.java

```diff
@@ -124,7 +130,61 @@ public void testEnableMtom() {
 
     private CONNECTClient<TestServicePortType> createClient(String url, AssertionType assertion) {
         return CONNECTClientFactory.getInstance().getCONNECTClientUnsecured(new TestServicePortDescriptor(), url,
-                assertion);
+            assertion);
+    }
+
+    @Test
+    public void testHttpsClientDisableCNCheckTrue() {
+        String url = "https://url";
+        AssertionType assertion = new AssertionType();
+        assertion.setTransactionTimeout(-1);
+
+        CONNECTClient<TestServicePortType> endpoint = null;
+        try {
+            PropertyAccessor.getInstance().setProperty(NhincConstants.GATEWAY_PROPERTY_FILE,
+                NhincConstants.DISABLE_CN_CHECK, "true");
+
+            boolean disableCNCheck = PropertyAccessor.getInstance()
+                .getPropertyBoolean(NhincConstants.GATEWAY_PROPERTY_FILE, NhincConstants.DISABLE_CN_CHECK);
+
+            endpoint = createClient(url, assertion);
+
+            Client client = ClientProxy.getClient(endpoint.getPort());
+
+            HTTPConduit conduit = (HTTPConduit) client.getConduit();
+            TLSClientParameters tlsPara = conduit.getTlsClientParameters();
+
+            Assert.assertTrue("disableCNCheck==true", tlsPara.isDisableCNCheck());
+        } catch (PropertyAccessException ex) {
+            System.out.println("Unable to set in memory property for disable-CN-Check.");
```

Comment:
Fail the test here if we get to the catch. Test will pass even if the very first line on 144 throws the exception.